save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/utils/GitUtil.kt

```diff
@@ -88,14 +93,14 @@ fun pullProject(
             .setFastForward(MergeCommand.FastForwardMode.FF)
             .call()
     } catch (ex: RefNotAdvertisedException) {
-        log.error("Provided branch $branchName seems to be an detached commit, pull command won't be performed!")
-        return null
+        git.close()
+        return Result.failure(RefNotAdvertisedException("Provided branch $branchName seems to be an detached commit, pull command won't be performed!"))
```

Comment:
What is the original error message in `ex`? Would it be better to wrap it into something like `IllegalArgumentException` with custom message and original exception as `cause`?

Reply:
They talking about, that `This Ref may not exist in the remote or may be hidden by permission settings`

I changed to `IllegalArgumentException `, for some reasons, when I used `cause` it prints nothing, so I put `ex` into message too

Reply:
Please put it as `cause`, just for consistency, too.